src/Workspaces/CSharp/Portable/Extensions/AttributeArgumentSyntaxExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions
+{
+    internal static class AttributeArgumentSyntaxExtensions
+    {
+        /// <summary>
+        /// Returns the parameter to which this argument is passed. If <paramref name="allowParams"/>
+        /// is true, the last parameter will be returned if it is params parameter and the index of
+        /// the specified argument is greater than the number of parameters.
+        /// </summary>
+        public static IParameterSymbol DetermineParameter(
+            this AttributeArgumentSyntax argument,
+            SemanticModel semanticModel,
+            bool allowParams = false,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var argumentList = argument.Parent as AttributeArgumentListSyntax;
+            if (argumentList == null)
+            {
+                return null;
+            }
+
+            var invocableExpression = argumentList.Parent as AttributeSyntax;
+            if (invocableExpression == null)
+            {
+                return null;
+            }
+
+            var symbol = semanticModel.GetSymbolInfo(invocableExpression, cancellationToken).Symbol;
+            if (symbol == null)
+            {
+                return null;
+            }
+
+            var parameters = symbol.GetParameters();
+
+            // Handle named argument
+            if (argument.NameColon != null && !argument.NameColon.IsMissing)
+            {
+                var name = argument.NameColon.Name.Identifier.ValueText;
+                return parameters.FirstOrDefault(p => p.Name == name);
+            }
+
+            // Handle positional argument
```

Comment:
what if it is a "Name =" argument?  That would not map to a parameter at all, regardless of position.

Reply:
That's right. I addressed this in 514e4fd2:
* Added a test for a cast in a named argument
* added a <remarks> section explaining the handling of named arguments
* Added an return at the top of the method for named arguments.

Reply:
Thanks!